src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -270,12 +276,12 @@ internal async Task<IVsPathContext> CreatePathContextAsync(NuGetProject nuGetPro
                     }
                 }
             }
-            catch (Exception e) when (e is KeyNotFoundException || e is InvalidOperationException)
+            catch (ProjectNotRestoredException e)
             {
                 var projectUniqueName = NuGetProject.GetUniqueNameOrName(nuGetProject);
                 var errorMessage = string.Format(CultureInfo.CurrentCulture, VsResources.PathContext_CreateContextError, projectUniqueName, e.Message);
                 _logger.Value.LogError(errorMessage);
-                throw new InvalidOperationException(errorMessage, e);
+                throw new ProjectNotRestoredException(errorMessage, e);
```

Comment:
Granted that `ProjectNotRestoredException` extends `InvalidOperationException`, so customers calling this method can catch it with `catch (InvalidOperationException)`. But, the XMLDocs for `IVsPathContextProvider` only list `InvalidOperationException` as a possible exception, not `ProjectNotRestoredException`.

I can't decide how I feel about this. It's not obvious to me that the PR should be changed, but it's also not obvious to me that it's fine. 😕

Reply:
This is `CreatePathContextAsync`, but aren't customers concerned with what's thrown by `TryCreateContext`?

Reply:
You're 100% right. I missed that the only public method catches ProjectNotRestoredException and then rethrows InvalidOperationException. All the methods that were changed to throw ProjectNotRestoredException are internal.